### PR TITLE
Use page title as default bookmark name

### LINKED ISSUE #966: bookmarks: Pass current page title to the create bookmark component automatically and autofil the bookmark name with it
> Can you check if it's easy to pass the title of the current page too so that it can be used in the `name` field as a default value?
> Please ignore if it requires more effort, we can convert this comment as an issue and deal with it in the future. 

_Originally posted by @manojVivek in https://github.com/responsively-org/responsively-app/pull/951#discussion_r1226755495_

### PR DESCRIPTION
**Pull request by PeterKwesiAnsah**
# ✨ Pull Request

### 📓 Referenced Issue
fixes #966

### ℹ️ About the PR
This PR uses the page title as default bookmark name by waiting for the dom to be ready on a primary webview, dispatching 
a action to add the page title to the redux store. This PR also improves the codebase by removing unnecessary state and effects

### 🖼️ Testing Scenarios / Screenshots

Screenshot (31)

### CODE REVIEW

**Comment by themohammadsa** [comment]
Hey @PeterKwesiAnsah!

Thank you for your swift contribution to this issue. Your solution has been impressive & it appears to be working as expected. 👍

However, I'd like to discuss an additional modification to the bookmark functionality. It seems that you have made a UX change that is not required. Allow me to explain the desired alteration below.

Currently, when a user clicks on the bookmark icon while it's already bookmarked, a fly-out appears, providing the user with the option to modify the name or URL of the bookmark.

Proposed Change:
Instead of removing the fly-out entirely and directly removing the bookmark, please modify your change so that the user experience remains the same as before.

Feel free to reach out if you have any more questions. 😄

**Review by themohammadsa — CHANGES_REQUESTED** [review]
Hey @PeterKwesiAnsah!

Thank you for your swift contribution to this issue. Your solution has been impressive & it appears to be working as expected. 👍

However, I'd like to discuss an additional modification to the bookmark functionality. It seems that you have made a UX change that is not required. Allow me to explain the desired alteration below.

Currently, when a user clicks on the bookmark icon while it's already bookmarked, a fly-out appears, providing the user with the option to modify the name or URL of the bookmark.

Proposed Change:
Instead of removing the fly-out entirely and directly removing the bookmark, please modify your change so that the user experience remains the same as before.

Feel free to reach out if you have any more questions. 😄

**Comment by PeterKwesiAnsah** [comment]
> Hey @PeterKwesiAnsah!
> 
> Thank you for your swift contribution to this issue. Your solution has been impressive & it appears to be working as expected. 👍
> 
> However, I'd like to discuss an additional modification to the bookmark functionality. It seems that you have made a UX change that is not required. Allow me to explain the desired alteration below.
> 
> Currently, when a user clicks on the bookmark icon while it's already bookmarked, a fly-out appears, providing the user with the option to modify the name or URL of the bookmark.
> 
> Proposed Change: Instead of removing the fly-out entirely and directly removing the bookmark, please modify your change so that the user experience remains the same as before.
> 
> Feel free to reach out if you have any more questions. 😄

Hi @themohammadsa , Thanks for the feedback. I will push some code

File: desktop-app/src/renderer/components/ToolBar/AddressBar/index.tsx [hunk 1]
```diff
@@ -241,7 +246,9 @@ const AddressBar = () => {
           >
             <Icon icon={isHomepage ? 'mdi:home' : 'mdi:home-outline'} />
           </Button>
-          <Bookmark currentAddress={address} />
+          {pageTitle && (
+            <Bookmark pageTitle={pageTitle} currentAddress={address} />
+          )}
```
Comment (themohammadsa) [inline desktop-app/src/renderer/components/ToolBar/AddressBar/index.tsx:251]: Is there any particular reason for the conditional check of `pageTitle` in this line? 
If not, then I think we can remove the conditional check and directly render the `<Bookmark pageTitle={pageTitle} currentAddress={address} />` component.

**Review by themohammadsa — CHANGES_REQUESTED** [review]
Everything is great @PeterKwesiAnsah, there's just a minor change that I have requested. Following this change, we can proceed to merge this PR. 

Kudos for your super-quick commits! 😁

File: desktop-app/src/renderer/components/ToolBar/AddressBar/index.tsx (hunk 1 above)
Comment (PeterKwesiAnsah) [reply desktop-app/src/renderer/components/ToolBar/AddressBar/index.tsx:251]: @themohammadsa Currently there's no reason, so Yes we can remove that.
Thanks

**Review by themohammadsa — APPROVED** [review]
Fantastic @PeterKwesiAnsah! Everything looks perfect to me. Merging this right away!  🚀

Hoping for more and more contributions from your end in the future! 😄

**Pull request merged**